client/reducers/cycles.js

```diff
@@ -1,26 +1,46 @@
-import { CYCLES, DELETE_CYCLE } from '../actions/types';
+import {
+    CYCLES,
+    DELETE_CYCLE,
+    SAVE_CYCLE,
+    TEST_SUITE_VERSIONS
+} from '../actions/types';
 
 // TODO: Change this into a object key'd by cycle id?
 const initialState = {
-    cycles: []
+    cycles: [],
+    testSuiteVersions: []
 };
 
 export default (state = initialState, action) => {
     switch (action.type) {
+        case TEST_SUITE_VERSIONS: {
+            const testSuiteVersions = action.payload;
+            return Object.assign({}, state, {
+                testSuiteVersions
+            });
+        }
         case CYCLES: {
             const cycles = action.payload;
-            return {
+            return Object.assign({}, state, {
                 cycles
-            };
+            });
         }
         case DELETE_CYCLE: {
             const { cycleId } = action.payload;
             let remainingCycles = state.cycles.filter(
                 cycle => cycle.id !== cycleId
             );
-            return {
+            return Object.assign({}, state, {
                 cycles: remainingCycles
-            };
+            });
+        }
+        case SAVE_CYCLE: {
+            const cycle = action.payload;
+            let newCycleList = [...state.cycles];
+            newCycleList.unshift(cycle);
+            return Object.assign({}, state, {
+                cycles: newCycleList
+            });
```

Comment:
I recommend either putting `testSuiteVersions` in its own reducer file, or changing the structure of this file to not rely on default exports: 

```js
import {
    CYCLES,
    DELETE_CYCLE,
    SAVE_CYCLE,
    TEST_SUITE_VERSIONS
} from '../actions/types';

const initialCycleState = [];

export const cycles = (state = initialCycleState, action) => {
    switch (action.type) {
        case CYCLES: {
            const cycles = action.payload;
            return [
                ...cycles
            ];
        }
        case DELETE_CYCLE: {
            const { cycleId } = action.payload;
            let remainingCycles = state.cycles.filter(
                cycle => cycle.id !== cycleId
            );
            return [
                ...remainingCycles
            ];
        }
        case SAVE_CYCLE: {
            const cycle = action.payload;
            let newCycleList = [...state.cycles];
            newCycleList.unshift(cycle);
            return [
                ...newCycleList
            ];
        }
        default:
            return state;
    }
};

const initialTestSuiteVersionsState = [];

export const testSuiteVersions = (state = initialTestSuiteVersionsState, action) => {
    switch (action.type) {
        case TEST_SUITE_VERSIONS: {
            const testSuiteVersions = action.payload;
            return [
                ...testSuiteVersions
            ];
        }
        default:
            return state;
    }
};
```

Then in your `reducers/index.js`, you'd import the reducers by export name: 

```js
import { combineReducers } from 'redux';
import login from './login';
import {cycles, testSuiteVersions} from './cycles';
import users from './users';

export default combineReducers({
    login,
    cycles,
    testSuiteVersions,
    users
});
```